src/nvim/os/env.c

```diff
@@ -451,18 +455,38 @@ static char *vim_version_dir(const char *vimdir)
   return NULL;
 }
 
-/// If the string between "p" and "pend" ends in "name/", return "pend" minus
-/// the length of "name/".  Otherwise return "pend".
-static char *remove_tail(char *p, char *pend, char *name)
+/// If `dirname + "/"` precedes tail in the path, return the pointer to
+/// `dirname + "/" + tail`.  Otherwise return `tail`.
+///
+/// Examples (path = /usr/local/share/nvim/runtime/doc/help.txt):
+///
+///   tail    = help.txt
+///   dirname = doc
+///   -> doc/help.txt
+///
+///   tail    = doc/help.txt
+///   dirname = runtime
+///   -> runtime/doc/help.txt
+///
+///   tail    = runtime/doc/help.txt
+///   dirname = vim74
+///   -> runtime/doc/help.txt
+///
+/// @param path    Path to a file
+/// @param tail    A suffix of the path
+/// @param dirname The immediate path fragment before the tail
+/// @return The new tail including dirname or just tail
+static char *remove_tail(char *path, char *tail, char *dirname)
```

Comment:
The new param name `tail` confuses me, because `tail` is never "removed" from anything. Would a function name like `remove_before_tail` or `path_tail` be better? 

(`path_tail()` already exists in path.c. And, `remove_tail` has a rather narrow use-case so it probably shouldn't live in path.c.)